frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/create-sc.tsx

```diff
@@ -237,14 +238,21 @@ const CreateSC: React.FC<CreateSCProps> = ({ match, hasNoProvSC, mode, lsoNs })
     const { appName, ns } = match.params;
     try {
       setInProgress(true);
-      const { storageClass, encryption, nodes, enableMinimal } = state;
+      const { storageClass, encryption, nodes, enableMinimal, kms } = state;
       const storageCluster: StorageClusterKind = getOCSRequestData(
         storageClass,
         defaultRequestSize.BAREMETAL,
         encryption.clusterWide,
         enableMinimal,
+        '',
+        '',
+        kms.hasHandled && encryption.advanced,
       );
-      await Promise.all(labelNodes(nodes)).then(() => k8sCreate(OCSServiceModel, storageCluster));
+      const promises: Promise<K8sResourceKind>[] = [...labelNodes(nodes)];
+      if (encryption.advanced && kms.hasHandled) {
+        promises.push(...kmsResources(kms));
+      }
+      await Promise.all(promises).then(() => k8sCreate(OCSServiceModel, storageCluster));
```

Comment:
I understand its not the part of PR, but this is a minor :)
```suggestion
      await Promise.all(promises).then(() => k8sCreate(StorageClusterModel, storageCluster));
```

Reply:
No worries to take in another PR, this needs fix at various places.